sys/usb/usbus/cdc/ecm/cdc_ecm_netdev.c

```diff
@@ -59,6 +59,11 @@ static int _send(netdev_t *netdev, const iolist_t *iolist)
     uint8_t *buf = cdcecm->ep_in->ep->buf;
     const iolist_t *iolist_start = iolist;
     size_t len = iolist_size(iolist);
+    /* interface with alternative function ID 1 is the interface containing the
+     * data endpoints, no sense trying to transmit data if it is not active */
+    if (cdcecm->active_iface != 1) {
```

Comment:
What if we use CDC/ECM with another interface like CDC/ACM at the same time ? Will it works ?

Reply:
Yes

This state variable tracks the alternative interface selected by the host. When looking at the descriptor, there should be two interface descriptors for the CDC ECM, the first without bulk endpoints and the second with bulk endpoints. They both have the same interface number but different alternate settings values.

On enumeration the default interface is the one with alternate setting 0, when the host enables the link it changes the interface to alternate setting 1, this only changes that interface and none of the others, so any other interface available such as the CDC ACM interface is not modified by this.